src/services/CertificateGenerationService.ts

```diff
@@ -1,898 +1,900 @@
+                            };
+                        }),
+                    };
+                }
+            )
+            .catch((error: AWSError | Error) => {
+                console.log(error);
+            });
     }
-    if (!isSingleRecord(techRecords) && testResult.trailerId) {
-      console.log(
-        "No unique Tech Record found for vrm ",
-        testResult.vrm,
-        ". Trying TrailerID"
-      );
-      techRecords = await this.queryTechRecords(testResult.trailerId);
+
+    /**
+     * Method for getting make and model based on the vehicle from a test-result
+     * @param testResult - the testResult for which the tech record search is done for
+     */
+    public getVehicleMakeAndModel = async (testResult: any) => {
+        const searchRes = await this.callSearchTechRecords(testResult.vin);
+        const techRecord = await this.processGetCurrentProvisionalRecords(searchRes);
+        // Return bodyMake and bodyModel values for PSVs
+        return techRecord?.techRecord_vehicleType === VEHICLE_TYPES.PSV ? {
+            Make: (techRecord as TechRecordType<"psv">).techRecord_chassisMake,
+            Model: (techRecord as TechRecordType<"psv">).techRecord_chassisModel
+        } : {
+            Make: (techRecord as TechRecordType<"hgv" | "trl">).techRecord_make,
+            Model: (techRecord as TechRecordType<"hgv" | "trl">).techRecord_model
+        };
     }
-    // @ts-ignore - already handled undefined case.
-    if (!isSingleRecord(techRecords) || !techRecords[0].techRecord) {
-      console.error(
-        `Unable to retrieve unique Tech Record for Test Result:`,
-        testResult
-      );
-      throw new Error(`Unable to retrieve unique Tech Record for Test Result`);
+
+    /**
+     * Used to return a subset of technical record information.
+     * @param searchIdentifier
+     */
+    public callSearchTechRecords = async (searchIdentifier: string) => {
+        const config: IInvokeConfig = this.config.getInvokeConfig();
+        const invokeParams: InvocationRequest = {
+            FunctionName: config.functions.techRecordsSearch.name,
+            InvocationType: "RequestResponse",
+            LogType: "Tail",
+            Payload: JSON.stringify({
+                httpMethod: "GET",
+                path: `/v3/technical-records/search/${searchIdentifier}`,
+                pathParameters: {
+                    searchIdentifier
+                },
+            }),
+        };
+        return await this.lambdaClient.invoke(invokeParams)
+            .then(async (response) => {
+                try {
+                    const res = await this.lambdaClient.validateInvocationResponse(response);
+                    return JSON.parse(res.body);
+                } catch (e) {
+                    console.log("in search tech record catch block");
+                    console.log(e);
+                    return undefined;
+                }
+            });
```

Comment:
```suggestion
  try {
    const lambdaResponse= await this.lambdaClient.invoke(invokeParams);
    const res = await this.lambdaClient.validateInvocationResponse(lambdaResponse);
    return JSON.parse(res.body);
  } catch (e) {
    console.log("Error searching technical records")
    console.log(JSON.stringify(e))
    return undefined;
  }
```
The use of both `.then` and `async/await` syntax in a bit confusing.